src/components/Items/Items.js

```diff
@@ -92,18 +109,53 @@ export class Items extends Component {
       storeBlock = (
         <h3>
           {store.name}
+
           {address}
         </h3>
       );
     }
+    let header;
+    let editForm = (
+      <form onSubmit={e => this.newName(e)}>
+        <input
+          type="text"
+          ref={editListName => {
+            this.editListName = editListName;
+          }}
+        />
+        <button>
+          <img
+            className="editIcon"
+            src="/edit.png"
+            alt="editList"
+            type="submit"
+          />
+        </button>
+      </form>
+    );
+    if (editingName) {
+      header = <div>{editForm}</div>;
```

Comment:
You should really integrate this into the existing header. One way is to have a “displayName” variable that is either a `h1` or a `form` element depending on whether `editingName` is true/false.